kube/subscription-test.sh

```diff
@@ -0,0 +1,37 @@
+#/bin/bash
+
+echo "Testing subscriptions..."
+kubectl delete -f .
+echo "Waiting 20s for deletion..."
+sleep 20
+
+# Ensure deployments exist
+echo "Applying deployments..."
+kubectl apply -f server.yaml
+kubectl apply -f streaming.yaml
+
+# Scale deployments
+echo "Scaling deployments..."
+kubectl scale deployment test-redis-ws --replicas=3 streaming-client --replicas=3
+
+echo "Waiting 20s for server and clients to start..."
+sleep 20
+
+# NOW start writer job after everything else is ready
+echo "Starting writer job..."
```

Comment:
Is there a more idiomatic way to run a one-shot job than deleting and re-applying it?

Reply:
The best method that I found is to add a TTL to the job so that they automatically get cleaned up. I set it to be cleaned up after 60s, which means we have 60s to get the logs. There is some information about it here: https://kubernetes.io/docs/concepts/workloads/controllers/job/